File: 06_inheritance_in_ruby/lib/models/Animal.rb
class Animal

  attr_reader :name, :catchphrase

  def initialize(name, catchphrase)
    @name = name
    @catchphrase = catchphrase

    @@all << self
  end

  @@all = []

  def self.all
    @@all
  end

  def eat
    puts "Nom nom"
  end

end

File: 34-object-oriented-js/scripts/models/animal.js
class Animal {

  constructor(name, canBePet, diet) {
    this.name = name
    this.canBePet = canBePet
    this.diet = diet

    const animalElement = document.createElement('p')
    animalElement.innerText = `Name: ${this.name}`
    document.querySelector('#animal-container').append(animalElement)
  }

  eat() {
    return `I am eating ${this.diet} nom nom`
  }

  makeSound = function() {
    console.log(`${this.name} made a sound`)
  }

}
